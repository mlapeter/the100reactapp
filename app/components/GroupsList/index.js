// @flow
import autobind from "autobind-decorator";
import * as React from "react";
import { FlatList, Image, View } from "react-native";

import ImageCard from "../ImageCard";
import defaultUserHeaderBackground from "../../assets/images/d2-all.jpg";

type GroupsListProps = {
  groups?: array
};

export default class GroupsList extends React.Component<GroupsListProps<>> {
  @autobind
  renderItem(group: string): React.Node {
    const navigation = this.props.navigation;
    let picture =
      group.header_background_image_api === "img/default-group-header.jpg"
        ? defaultUserHeaderBackground
        : { uri: picture };

    return (
      <ImageCard
        title={group.item.name}
        picture={defaultUserHeaderBackground}
        heightRatio={0.4}
        onPress={() => navigation.navigate("Group", { groupId: group.item.id })}
      />
    );
  }

  render() {
    return (
      <FlatList
        data={this.props.groups}
        renderItem={this.renderItem}
        keyExtractor={(item, index) => index.toString()}
        refreshing={this.props.isLoading}
      />
    );
  }
}
