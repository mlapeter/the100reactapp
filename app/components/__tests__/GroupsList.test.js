import React from "react";
import { View } from "react-native";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import GroupsList from "../GroupsList";

const groups = [
  {
    header_background_image_api: "img/default-group-header.jpg",
    id: 1578,
    name: "The100 Lounge",
    nickname: "",
    users_count: 77
  }
];

it("renders successfully without children", () => {
  const rendered = renderer.create(<GroupsList groups={[]} />).toJSON();
  console.log(rendered);
  expect(rendered).toBeTruthy();
});

it("renders successfully with one group", () => {
  const rendered = renderer.create(<GroupsList groups={groups} />).toJSON();
  console.log(rendered);
  expect(rendered).toBeTruthy();
});

it("renders with one group", () => {
  const rendered = shallow(<GroupsList groups={groups} />);
  expect(rendered.dive()).toMatchSnapshot();
});
