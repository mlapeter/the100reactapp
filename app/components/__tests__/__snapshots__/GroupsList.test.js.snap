// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with one group 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlatList
    data={
      Array [
        Object {
          "header_background_image_api": "img/default-group-header.jpg",
          "id": 1578,
          "name": "The100 Lounge",
          "nickname": "",
          "users_count": 77,
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "data": Array [
        Object {
          "header_background_image_api": "img/default-group-header.jpg",
          "id": 1578,
          "name": "The100 Lounge",
          "nickname": "",
          "users_count": 77,
        },
      ],
      "disableVirtualization": false,
      "getItem": [Function],
      "getItemCount": [Function],
      "horizontal": false,
      "initialNumToRender": 10,
      "keyExtractor": [Function],
      "maxToRenderPerBatch": 10,
      "numColumns": 1,
      "onEndReachedThreshold": 2,
      "renderItem": [Function],
      "scrollEventThrottle": 50,
      "updateCellsBatchingPeriod": 50,
      "viewabilityConfigCallbackPairs": Array [],
      "windowSize": 21,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": Array [
          Object {
            "header_background_image_api": "img/default-group-header.jpg",
            "id": 1578,
            "name": "The100 Lounge",
            "nickname": "",
            "users_count": 77,
          },
        ],
        "disableVirtualization": false,
        "getItem": [Function],
        "getItemCount": [Function],
        "horizontal": false,
        "initialNumToRender": 10,
        "keyExtractor": [Function],
        "maxToRenderPerBatch": 10,
        "numColumns": 1,
        "onEndReachedThreshold": 2,
        "renderItem": [Function],
        "scrollEventThrottle": 50,
        "updateCellsBatchingPeriod": 50,
        "viewabilityConfigCallbackPairs": Array [],
        "windowSize": 21,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
